### ISSUE TITLE
Compiler error for py_vector.cpp

### ISSUE THREAD

**Issue by prashantraina** [libigl/libigl#825]
Hi, this line in pyvector.cpp (line 644) is giving a compiler error.
I was able to work around it by simply commenting it out.

```
/* Bindings for MatrixXb */
bind_eigen_2<Eigen::Matrix<bool,Eigen::Dynamic,Eigen::Dynamic> > (me, "MatrixXb");
```

The error was a static assert fail from Eigen saying YOU_MIXED_DIFFERENT_NUMERIC_TYPES__YOU_NEED_TO_USE_THE_CAST_METHOD_OF_MATRIXBASE_TO_CAST_NUMERIC_TYPES_EXPLICITLY

I'm using GCC 8 on Arch Linux.

**Comment by phg1024** [comment]
Any update on this issue?

**Comment by jdumas** [comment]
New python bindings are in the work in the `dev-python` branch, so hopefully this issue will be obsolete when they come out.

**Comment by fwilliams** [comment]
We're currently in the process of rewriting the Python bindings from the ground up using [NumpyEigen](https://github.com/fwilliams/numpyeigen) to be fully pythonic , have zero copy overhead to/from Python and C++. Check out the `dev-python` branch to see the progress. We should release a completed version in a month or two. 

As such, we're not going to fix any bugs in the old bindings since they will be removed.

**Comment by phg1024** [comment]
Thanks!
btw I managed to build the python bindings using the following method: 
1. Modify `CMakeLists.txt` under `python` folder to make sure we use the Eigen version under `external`
2. Build libigl
```
cd $LIBIGL_PATH
mkdir build
cd build
cmake .. -DCMAKE_BUILD_TYPE=Release
make -j10
```
This can build the python bindings successfully. The error I got is actually due to a different Eigen version existing in my system, so pointing to the "right" Eigen version resolves this issue.

**Comment by fwilliams** [comment]
If you send a PR with these changes and it doesn't break the tests we can merge it.

**Comment by phg1024** [comment]
Added #995. I feel the fix is pretty hacky though.

After compiling igl a few times, on several platforms, I found building igl from the root directory already creates a `pyigl.so` file under python. However, I ran into some other issues that actually forced me to use `setup.py` to build the python bindings: I was using conda+python3.5 on MacOS Mojave, and for some reason several symbols (for example, `inner_angle_XXX`) are missing and I could not import `pyigl`. I checked that I was using the correct python interpreter and libraries. Didn't spend too much time to look into this but I think this is a compiler dependent issue since the `pyigl.so` file built under ubuntu does not have this issue on my side. Anyway, the changes in the PR fixes this issue.
